jobrunner/project.py

```diff
@@ -239,6 +239,23 @@ def add_config_to_run_command(run_command, config):
     return f"{run_command} --config '{config_as_json}'"
 
 
+def requires_db_access(args):
+    """
+    By default actions do not have database access, but certain trusted actions require it
+    """
+    valid_commands = {
+        "cohortextractor": ("generate_cohort", "generate_codelist_report"),
+        "cohortextractor-v2": ("generate_cohort", "generate_dataset"),
+        "databuilder": ("generate_dataset"),
```

Comment:
This looks like a single-element tuple but it isn't!

I think you want:

```
         "databuilder": ("generate_dataset",),
```

Reply:
Well ingl-spied! Missed by three reviewers (incl. myself) I think. The perennial Python footgun!

Reply:
New PR over here https://github.com/opensafely-core/job-runner/pull/380